io/src/lib.rs

```diff
@@ -0,0 +1,70 @@
+#![no_std]
+
+use codec::{Decode, Encode};
+
+use gstd::ActorId;
+use scale_info::TypeInfo;
+
+
+#[derive(Debug, Default, Encode, Decode, TypeInfo, Clone)]
+pub struct IcoState {
+    pub ico_started: bool,
+    pub start_time: u64,
+    pub duration: u64,
+    pub ico_ended: bool,
```

Comment:
Aren't `ico_started` and `ico_ended` excessive if we already have  `start_time` and `duration`?

Reply:
We need `ico_ended` because the contract can be ended before `start_time + duration` (if all tokens have been sold).
`ico_started ` can be replaced with `if start_time != 0` but it seems to me that this way the code becomes more easily understandable (If I'm wrong, I'll fix it)